ops/model.py

```diff
@@ -1182,8 +1183,17 @@ def __init__(
         rotation: Optional[SecretRotate],
         rotates: Optional[datetime.datetime],
         description: Optional[str] = None,
+        *,
+        model_uuid: Optional[str] = None,
     ):
-        self.id = Secret._canonicalize_id(id)
+        if model_uuid is None:
+            warnings.warn(
+                '`model_uuid` should always be provided when creating a '
```

Comment:
In what cases can this happen? Only if people manually create a `Secret` object? (Which they shouldn't be doing.)

Reply:
> In what cases can this happen? Only if people manually create a `Secret` object? (Which they shouldn't be doing.)

Manually create a `SecretInfo`, yes. And yes, I don't know why they would, although we don't tell them not to (as we do with `Secret`, for example). This is probably overly cautious - I can't find any charm doing this (Scenario does, but we can fix that in advance of a release), but seemed better than just having `None` work and then in some future version (3.0?) suddenly not work.